workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -129,5 +130,214 @@ private static bool TryGetApkPath(string rootPath, out string apkPath)
+                    }
+                    else
+                    {
+                        var spatialKeyNode = doc.CreateNode("element", "key", string.Empty);
+                        spatialKeyNode.InnerText = LaunchArguments.iOSEnvironmentKey;
+                        var spatialValueNode = doc.CreateNode("element", "string", string.Empty);
+                        spatialValueNode.InnerText = arguments;
+                        envValueNode.AppendChild(spatialKeyNode);
+                        envValueNode.AppendChild(spatialValueNode);
+                    }
+
+                    break;
+                }
+
+                doc.Save(xcTestRunPath);
+
+                // fix document after updating it
+                var text = File.ReadAllText(xcTestRunPath);
+                text = text.Replace("[]>", ">");
+                File.WriteAllText(xcTestRunPath, text);
+
+                if (availableDevices.Count > 0)
+                {
+                    deviceUID = availableDevices[0].Trim('[', ']');
+                }
+                else
+                {
+                    // start simulator
+                    deviceUID = toolsConfig.GetSimulatorUID().Trim('[', ']');
+
+                    if (RedirectedProcess.Command("xcrun")
+                        .WithArgs("instruments", "-w", deviceUID, "-t", "Blank")
+                        .Run() != 0)
+                    {
+                        Debug.LogError("Was unable to start iOS Simulator.");
+                        return;
+                    }
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing your app", 0.7f);
+                var command = "osascript";
+                var commandArgs = $@"-e 'tell application ""Terminal""
+                                     activate
+                                     do script ""xcodebuild test-without-building -destination id={deviceUID} -xctestrun {xcTestRunPath}""
+                                     end tell'";
+
+                var processInfo = new ProcessStartInfo(command, commandArgs)
+                {
+                    CreateNoWindow = false,
+                    UseShellExecute = true,
+                    WorkingDirectory = Common.SpatialProjectRootDir
+                };
+
+                var process = Process.Start(processInfo);
+
+                if (process == null)
+                {
+                    Debug.LogError("Failed to start app on iOS Simulator.");
+                    return;
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        private static bool TryGetXCTestRunPath(bool forSimulator, out string xctestrunPath)
+        {
+            foreach (var file in Directory.GetFiles(DerivedDataPath, "*.xctestrun", SearchOption.AllDirectories))
```

Comment:
Small nit: `file` -> `fileName`/`filePath` so the `file.Contains()` call is clearer (one could interpret that as looking into the file)